### PR TITLE
Milestone 3: Create class association

### PR DESCRIPTION
**Pull request by Profsain**
I completed this part of the project by implementing the following requirement
- Created a class classroom with initialize constructor
- Added getter and setter method for classroom instance variable
- Created has-many/belongs-to relationship between Classroom and Student

- Created a class book with an instance variable @title, @author, with initialize constructor
- Added getter and setter method

- Created a class rental with an instance variable @date and initialize constructor
- Added getter and setter method
- Created the many-to-many (also has-many-through) relationship between Person and Book using the intermediate class Rental

### CODE REVIEW

File: book.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+require './rental'
+
+class Book
+  attr_accessor :title, :author, :rental
```
Comment (TedLivist) [inline book.rb]: - [ ] Kindly correct your type error here. It should be `rentals`, with the 's'.

File: rental.rb [hunk 2]
```diff
@@ -0,0 +1,17 @@
+require './person'
+require './book'
+
+class Rental
+  attr_reader :person, :book
+  attr_accessor :date
+
+  def initialize(date)
```
Comment (TedLivist) [inline rental.rb]: - [ ] To be able to set up the has many through association, you still need to provide the `book` and `person` objects.

  ```
  def initialize(date, book, person)
    ...
  ```

File: person.rb [hunk 3]
```diff
@@ -1,20 +1,22 @@
 require './main'
 require './capitalize_decorator'
 require './trimmer_decorator'
+require './rental'
 
 class Person < Nameable
+  # getter and setter
+  attr_reader :id
+  attr_accessor :name, :age, :rental
```
Comment (TedLivist) [inline person.rb]: - [ ] This should be `rentals`

File: person.rb [hunk 4]
```diff
@@ -1,20 +1,22 @@
 require './main'
 require './capitalize_decorator'
 require './trimmer_decorator'
+require './rental'
 
 class Person < Nameable
```
Comment (TedLivist) [inline person.rb:6]: - [ ] Your Person class is missing the other end of the has many relationship. You need to add a method for that:

  ```
  def add_rental(book, date)
    Rental.new(date, book self)
  end
  ```

File: book.rb [hunk 5]
```diff
@@ -0,0 +1,11 @@
+require './rental'
+
+class Book
```
Comment (TedLivist) [inline book.rb:3]: - [ ] Your Book class is missing the other end of the has many through association. Kindly add a method like so:

  ```
  def add_rental(person, date)
    Rental.new(date, self, person)
  end
  ```

**Review by TedLivist — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ ♻️

Hi @Profsain,

Good job so far!
✔️ Good documentation
✔️ Establishment of association between classroom and student
However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: book.rb (hunk 5 above)
Comment (Profsain) [reply book.rb:3]: Thank you for the feedback. I will implement all changes.

**Review by FrederickMih — APPROVED** [review]
# Approved :heavy_check_mark: 

Hi @Profsain , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights

- PR is professional :heavy_check_mark: 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**